### ISSUE TITLE
Wrong version of npm module "debug" added to bundle due to "node process"

### ISSUE THREAD

**Issue by bitflower** [ionic-team/stencil#494]
**Bug repo:**
[https://github.com/bitflower/stencil-debug-problem](https://github.com/bitflower/stencil-debug-problem)

**Current behavior:**
See repo: importing "debug" doesn't work. I guess to to the fact that the Stencil compiler runs "on the server" (=node/terminal) and the debug module returns the node.js insetad of the browser.js because of that.

**Expected behavior:**
We can use "debug".

**Steps to reproduce:**
See repo

**Comment by jthoms1** [comment]
Could you please update stencil to the latest release? `npm install @stencil/core@latest`  I believe this was fixed in 0.3.0

**Comment by bitflower** [comment]
Hey guys, sorry for being late to the party.

I'm on Stencil 0.4.3 (can't go higher to 0.6.x right now).

But anyway it is not fixed:
<img width="1014" alt="bildschirmfoto 2018-02-23 um 16 04 53" src="https://user-images.githubusercontent.com/6666636/36600762-72363ec2-18b3-11e8-97c8-b2af17b299f5.png">

Chrome is trying to access `process.stderr.fd`....

Re-open or new issue?

Sorry for the inconvenience!

**Comment by jthoms1** [comment]
I pulled down your repo and updated to the latest stencil.  Then changed the debug import to as follows.

`import debug from 'debug';`

The issue appears to be fixed no build or runtime error.

**Comment by bitflower** [comment]
Hey @jthoms1, thanks for the continuous support (that said upfront ;-) ).
I work on different things and therefore always might need some time to get back..

Anyway, I had a closer look to the problem again and I created the error scenario in the repo. If you might want to have a look that would be awesome!

Repo: [bitflower/stencil-debug-problem](https://github.com/bitflower/stencil-debug-problem)

Still gives me this:
<img width="957" alt="bildschirmfoto 2018-03-02 um 12 38 23" src="https://user-images.githubusercontent.com/6666636/36897489-fe901484-1e16-11e8-8eac-09d41e744dfd.png">

Would love to se this re-opened.

Let me know if I can provide anything else!

Node: 8.9.4
OS: macOS 10.13.3

**Comment by bitflower** [comment]
Updated the repo to Stencil 0.7.0....

Problem persists :-(

**Comment by kirillgroshkov** [comment]
Same here. Going same path as @bitflower :)

**Comment by bitflower** [comment]
Hey @jthoms1 I have created a (private) repo for #618 which also showcases this issue.

I have added you to the collaborators! Hope it helps!

Repo:
[https://github.com/bitflower/caseos-ui/tree/data_component](https://github.com/bitflower/caseos-ui/tree/data_component)

**Comment by bitflower** [comment]
Had the tiny hope of this being fixed after #618 is working now :-)

Do you need something else I can provide?

**Comment by bitflower** [comment]
Hey guys, this keeps popping up everytime I add a module using `debug` (just today). How come nobody else has this!? Is it my setup? Hm ...

**Comment by jthoms1** [comment]
@bitflower I will spend some extra time on this today.  Hopefully I can find you a solution.

**Comment by jthoms1** [comment]
It appears that there is a merged PR in `debug` that should fix this issue, but it has not been released yet. https://github.com/visionmedia/debug/pull/507

**Comment by jthoms1** [comment]
I am going to close this issue because it should be fixed in the next release of `debug`.

**Comment by bitflower** [comment]
Hey @jthoms1! Great news, thanks a bunch! Am working around this in the meantime....

Wow, `debug` has kinda a slow release intervals compared to @stencil LOL
